workflow/executor/executor.go

```diff
@@ -1005,11 +994,11 @@ func (we *WorkflowExecutor) Wait(ctx context.Context) error {
 	annotationUpdatesCh := we.monitorAnnotations(ctx)
 	go we.monitorDeadline(ctx, annotationUpdatesCh)
 
-	_ = wait.ExponentialBackoff(ExecutorRetry, func() (bool, error) {
-		err = we.RuntimeExecutor.Wait(ctx, mainContainerID)
+	err = wait.ExponentialBackoff(ExecutorRetry, func() (bool, error) {
+		err := we.RuntimeExecutor.Wait(ctx, mainContainerID)
 		if err != nil {
 			log.Warnf("Failed to wait for container id '%s': %v", mainContainerID, err)
-			return false, err
+			return false, nil
```

Comment:
big change here

Reply:
previously this code would have NEVER retried

Reply:
After this exponentialBackoff this function needs to return the original error instead of `ErrWaitTimeout`